File: app/src/main/java/de/txserver/slickupnp/upnp/ContentDirectoryBrowseCallbacks.java
package de.txserver.slickupnp.upnp;

import java.util.ArrayList;

import de.txserver.slickupnp.helper.DeviceModel;
import de.txserver.slickupnp.helper.ItemModel;

public interface ContentDirectoryBrowseCallbacks {

    void setShowRefreshing(boolean show);
    void onDisplayDevices();
    void onDisplayDirectories();
    void onDisplayItems(ArrayList<ItemModel> items);
    void onDisplayAddItems(ArrayList<ItemModel> items);
    void onDisplayItemsError(String error);
    void onDeviceAdded(DeviceModel device);
    void onDeviceRemoved(DeviceModel device);
}

File: app/src/main/kotlin/de/txserver/slickupnp/upnp/ContentDirectoryBrowseCallbacks.kt
package de.txserver.slickupnp.upnp

import java.util.ArrayList

import de.txserver.slickupnp.helper.DeviceModel
import de.txserver.slickupnp.helper.ItemModel

interface ContentDirectoryBrowseCallbacks {

    fun setShowRefreshing(show: Boolean)
    fun onDisplayDevices()
    fun onDisplayDirectories()
    fun onDisplayItems(items: ArrayList<ItemModel>)
    fun onDisplayAddItems(items: ArrayList<ItemModel>)
    fun onDisplayItemsError(error: String)
    fun onDeviceAdded(device: DeviceModel)
    fun onDeviceRemoved(device: DeviceModel)
}
